spec/features/score_spec.rb

```diff
@@ -0,0 +1,286 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Score', :js do
+  let(:exercise) { create(:hello_world) }
+  let(:contributor) { create(:external_user) }
+  let(:status) { 'ok' }
+  let(:scoring_response) do
+    {
+      users: {all: users_success + users_error + users_unsupported, success: users_success, error: users_error, unsupported: users_unsupported},
+      score: {original: score, sent: score_sent},
+      deadline:,
+      detailed_results: [],
+    }
+  end
+
+  let(:calculate_response) do
+    {
+      status:,
```

Comment:
In the case `when LTI outcomes are not supported`, you're setting this `status` to `unsupported`. However, I am not sure where this could ever happen within the actual code (when not using a mock). Do you have a pointer for me, please?

Reply:
It seems to be a mistake from my side. There only is `ok` and `failed` as status, right? In this unsupported case is the status then failed?

Reply:
> It seems to be a mistake from my side.

Alright.

> There only is ok and failed as status, right? 

In addition to these two (which I also see), there are a few more, but all related to runner management -- nothing in regard with LTI.

> In this unsupported case is the status then failed?

No. The score run can still succeed (Poseidon, ...), while the LTI submission could be unsupported (missing LTI params). Those are two different things...

I am currently making a few suggestions to your code, including a removal of this `status` variable. Please be patient for my upcoming changes :)